### PR TITLE
Use match instead of nested `if let`

### PR DESCRIPTION
**Pull request by Sh3Rm4n**
I stumbled upon this code again and thought I could make it a bit more readable.

### CODE REVIEW

File: src/rcc.rs [hunk 1]
```diff
@@ -438,32 +438,33 @@ impl CFGR {
     fn get_sysclk(&self) -> (u32, cfgr::SW_A, Option<PllConfig>) {
         // If a sysclk is given, check if the PLL has to be used,
         // else select the system clock source, which is either HSI or HSE.
-        if let Some(sysclk) = self.sysclk {
-            if let Some(hseclk) = self.hse {
-                if sysclk == hseclk {
+        match (self.sysclk, self.hse) {
+            (Some(sysclk), Some(hse)) => {
+                if sysclk == hse {
```
Comment (ra-kete) [inline src/rcc.rs:443]: You could even put this in an if-guard on the match to remove one level of indentation

File: src/rcc.rs [hunk 2]
```diff
@@ -438,32 +438,33 @@ impl CFGR {
...
+                    (sysclk, cfgr::SW_A::HSE, None)
                 } else {
-                    let clock_with_pll = self.calc_pll(sysclk);
-                    (clock_with_pll.0, cfgr::SW_A::PLL, Some(clock_with_pll.1))
+                    let (sysclk, pll_config) = self.calc_pll(sysclk);
+                    (sysclk, cfgr::SW_A::PLL, Some(pll_config))
+                }
+            }
+            (Some(sysclk), None) => {
+                if sysclk == HSI {
```
Comment (ra-kete) [inline src/rcc.rs:456]: Same here, re if-guard

File: src/rcc.rs [hunk 3]
```diff
@@ -438,32 +438,33 @@ impl CFGR {
...
+                    // No need to use the PLL
+                    (sysclk, cfgr::SW_A::HSE, None)
```
Comment (ra-kete) [inline src/rcc.rs:458]: I haven't looked into the manual, but using `SW_A::HSE` here seems fishy. Shouldn't that be `SW_A::HSI`?

**Comment by ra-kete** [comment]
Looks like we got another one of these hanging Travis builds here.

File: src/rcc.rs (hunk 1 above)
Comment (Sh3Rm4n) [reply src/rcc.rs:443]: Good idea!

File: src/rcc.rs (hunk 3 above)
Comment (Sh3Rm4n) [reply src/rcc.rs:458]: Hm. :thinking: It was there before (though it was my addition before as well). It should be `SW_A::HSI`. I'll recheck against the reference manual.

File: src/rcc.rs (hunk 1 above)
Comment (Sh3Rm4n) [reply src/rcc.rs]: Nice, with the help of if-guard, I could simplify it quite a bit further!

**Pull request merged**